### PR TITLE
Raise `ValueError` when waiting trial is told

### PR DESCRIPTION
**Pull request by not522**
## Motivation
The told trial should be `RUNNING`, not `WAITING`.

## Description of the changes
Fix `_tell_with_warning` and add a test.

### CODE REVIEW

File: tests/study_tests/test_study.py [hunk 1]
```diff
@@ -1437,6 +1437,11 @@ def test_tell_invalid() -> None:
     with pytest.raises(ValueError):
         study.tell(study.ask().number + 1, 1.0)
 
+    # Waiting trial cannot be told.
+    with pytest.raises(ValueError):
+        study.enqueue_trial({})
+        study.tell(len(study.trials) - 1, 1.0)
```
Comment (toshihikoyanase) [inline tests/study_tests/test_study.py:1443]: A nit, but how about specifying trial number explicitly?
```suggestion
        study.tell(study.trials[-1].number, 1.0)
```

**Review by toshihikoyanase — COMMENTED** [review]
LGTM with a small comment. In my understanding, waiting trials will change the status to RUNNING when users call `Study.ask`, right?

**Comment by toshihikoyanase** [comment]
@knshnb Could you review this PR, please?

**Comment by not522** [comment]
> In my understanding, waiting trials will change the status to RUNNING when users call Study.ask, right?

Exactly. Only `ask` and `optimize` change the state to `RUNNING`.  (`optimize` calls `ask` internally.)

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your confirmation. LGTM.

**Review by knshnb — APPROVED** [review]
LGTM! Thanks for the PR.

**Pull request merged**